src/cli/pubkey.cpp

```diff
@@ -215,6 +217,15 @@ class PK_Sign final : public Command
          this->read_file(get_arg("file"), onData);
 
          output() << Botan::base64_encode(signer.signature(rng())) << "\n";
+
+         if(key->stateful_operation())
+            {
+            std::ofstream updated_key(key_file);
+            if(passphrase.empty())
+               updated_key << Botan::PKCS8::PEM_encode(*key);
```

Comment:
If the application crashes i.e. due to an out of memory error, power outage etc. between [L219](https://github.com/randombit/botan/blob/db1b06d2ee44a1bc72de3bdd4c4098cc8c149dcc/src/cli/pubkey.cpp#L219) and the key update here the signature might have been used or stored already but the key has not been updated. Once the application is restarted the next signature operation will reuse the leaf index breaking XMSS's security.